src/App.js

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from "react";
+import './App.css';
+import Board from './components/Board/Board';
+import BoardList from './components/BoardList/BoardList';
+import Sidebar from './components/Sidebar/Sidebar';
+import axios from 'axios';
+import SortOrder from "./components/SortOrder/SortOrder";
+
+const kBaseUrl = process.env.REACT_APP_BACKEND_URL_DEPLOYED;
+
+function App() {
+  const [boardData, setBoardData] = useState([]);
+  const [cardData, setCardData] = useState([])
+  const [selectedBoard, setSelectedBoard] = useState(undefined);
+  const [isSidebarShown, setIsSidebarShown] = useState(true)
+  const [selectedSort, setSelectedSort] = useState(undefined); 
+  
+  useEffect(() => {
+    axios
+      .get(`${kBaseUrl}/boards`)
+      .then((res) => setBoardData(res.data))
+    .catch((err) => console.log(err));
+}, []);
+
+
+  useEffect(() => {
+    if (selectedBoard) {
+      let url = `${kBaseUrl}/boards/${selectedBoard}/cards`; 
+
+      if (selectedSort) {
+        url += `?sort=${selectedSort}`;
+      }
+    axios
+      .get(url)
+      .then((res) => setCardData(res.data))
+      .catch((err) => console.log(err));
+  }
+}, [selectedBoard, selectedSort]);
+
+const handleBoardSubmit = (newBoardData) => {
+  axios
+    .post(`${kBaseUrl}/boards`, newBoardData)
+    .then((res) => {
+      setBoardData((prev) => [...prev, res.data.board]);
+    })
+    .catch((err) => console.log(`Failed to assign card to board. Error: ${err}`));
+};
+
+const chooseToggleButtonSize = 
+(toggleState) => {
+  let buttonStyle = 'toggleButtonToClose';
+
+  if (!toggleState) {
+    buttonStyle = 'toggleButtonToOpen';
+  } else {
+    buttonStyle = 'toggleButtonToClose';
+  }
+  return buttonStyle;
+}
+
+const handleCardSubmit = (newCardFormProps) => {
+  // Handle UI after card has been created from the form
+  return axios
+    .post(
+      `${kBaseUrl}/boards/${selectedBoard}/cards`,
+      { message: newCardFormProps.message },
+    )
+    .then((res) => {
+      console.log(`Assigned card to a boardId=${selectedBoard}. Response: ${res}`);
+      // UI change
+      setCardData((prev) => [...prev, res.data.card]);
+    })
+    .catch((error) => {
+      console.error(`Failed to assign card to board. Error: ${error}`);
+    });
+};
+
+const handleLike = id => {
+  return axios
+  .patch(`${kBaseUrl}/cards/${id}`)
+  .then((res) => {
+    setCardData((prev) => {
+      return prev.map((card) =>{
+        if (id === card.id) {
+          return {...card, likes_count: res.data.card.likes_count};
```

Comment:
We can replace the old card data in the list entirely with the result of the API call. This would allow the frontend to need to know nothing about what fields get updated for a like.
```py
          return res.data.card;
```